test/e2e/e2e.go

```diff
@@ -1357,187 +1388,216 @@ var _ = ginkgo.Describe("e2e multiple ecmp external gateway validation", func()
+		framework.Logf("the pod side node is %s and the source node ip is %s - %s", node.Name, addressesv4.nodeIP, addressesv6.nodeIP)
 
-		var (
-			pingSrc           string
-			ciNetworkFlag     = "{{ .NetworkSettings.Networks.kind.IPAddress }}"
-			srcPingPodName    = "e2e-exgw-ecmp-src-ping-pod"
-			command           = []string{"bash", "-c", "sleep 20000"}
-			testContainer     = fmt.Sprintf("%s-container", srcPingPodName)
-			testContainerFlag = fmt.Sprintf("--container=%s", testContainer)
-		)
+		ginkgo.By("Creating the source pod to reach the destination ips from")
+		clientPod := createGenericPod(f, srcPodName, node.Name, f.Namespace.Name, []string{"bash", "-c", "sleep 20000"})
+		addressesv4.srcPodIP, addressesv6.srcPodIP = getPodAddresses(clientPod)
+		framework.Logf("the pod source pod ip(s) are %s - %s", addressesv4.srcPodIP, addressesv6.srcPodIP)
 
-		// start the first container that will act as an external gateway
-		_, err := runCommand("docker", "run", "-itd", "--privileged", "--network", ciNetworkName, "--name", gwContainer1, "centos/tools")
-		if err != nil {
-			framework.Failf("failed to start external gateway test container %s: %v", gwContainer1, err)
-		}
-		// retrieve the container ip of the external gateway container
-		exGWIp1, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainer1)
-		if err != nil {
-			framework.Failf("failed to start external gateway test container: %v", err)
-		}
-		// trim newline from the inspect output
-		exGWIp1 = strings.TrimSuffix(exGWIp1, "\n")
-		if ip := net.ParseIP(exGWIp1); ip == nil {
-			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainer1, exGWIp1)
+		if addressesv6.srcPodIP != "" && addressesv6.nodeIP != "" {
+			testIPV6 = true
 		}
 
-		// start the second container that will act as an external gateway
-		_, err = runCommand("docker", "run", "-itd", "--privileged", "--network", ciNetworkName, "--name", gwContainer2, "centos/tools")
-		if err != nil {
-			framework.Failf("failed to start external gateway test container %s: %v", gwContainer2, err)
-		}
-		// retrieve the container ip of the external gateway container
-		exGWIp2, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainer2)
-		if err != nil {
-			framework.Failf("failed to start external gateway test container: %v", err)
+		// This sets up a listener that replies with the hostname, both on tcp and on udp
+		setupListeners := func(container, address string) {
+			cmd := []string{"docker", "exec", container, "bash", "-c", fmt.Sprintf("while true; do echo $(hostname) | nc -l -u %s %d | exit; done &", address, externalUDPPort)}
+			_, err = runCommand(cmd...)
+			framework.ExpectNoError(err, "failed to setup listener on ", address, container)
+
+			cmd = []string{"docker", "exec", container, "bash", "-c", fmt.Sprintf("while true; do echo $(hostname) | nc -l %s %d | exit; done &", address, externalTCPPort)}
```

Comment:
not a bash expert , 

```
"while true; do echo $(hostname) | nc -l %s %d | exit; done &
```
is not `|| exit`  ?

Reply:
Neither am I, but looking at that again I guess the exit can be removed totally.